pep-0684.rst

```diff
@@ -0,0 +1,640 @@
+of the things that have already been done.
+
+* cleanup of runtime initialization (see :pep:`432` / :pep:`587`)
+* extension module isolation machinery (see :pep:`384` / :pep:`3121` / :pep:`489`)
+* isolation for many builtin modules
+* isolation for many stdlib extension modules
+* addition of ``_PyRuntimeState``
+* no more ``_Py_IDENTIFIER()``
+* statically allocated:
+
+  * empty string
+  * string literals
+  * identifiers
+  * latin-1 strings
+  * length-1 bytes
+  * empty tuple
+
+Tooling
+-------
+
+As already indicated, there are several tools to help identify the
+globals and reason about them.
+
+* Tools/c-analyzer/cpython/globals-to-fix.tsv - the list of remaining globals
+* Tools/c-analyzer/c-analyzer.py
+
+  + analyze - identify all the globals
+  + check - fail if there are any unsupported globals that aren't ignored
+
+* Tools/c-analyzer/table-file.py - summarize the known globals
+
+As well, the check for unsupported globals is incorporated into CI so that
+no new globals are accidentally added.
+
+Global Objects
+--------------
+
+Global objects that are safe to be shared (without a GIL) between
+interpreters can stay on ``_PyRuntimeState``.  Not only must the object
+be effectively immutable (e.g. singletons, strings), but not even the
+refcount can change for it to be safe.  Immortality (:pep:`683`)
+provides that.  (The alternative is that no objects are shared, which
+adds significant complexity to the solution, particularly for the
+objects `exposed in the public C-API <capi objects_>`_.)
+
+Builtin static types are a special case of global objects that will be
+shared.  They are effectively immutable except for one part:
+``__subclasses__`` (AKA ``tp_subclasses``).  We expect that nothing
+else on a builtin type will change, even the content
+of ``__dict__`` (AKA ``tp_dict``).
+
+``__subclasses__`` for the builtin types will be dealt with by making
+it a getter that does a lookup on the current ``PyInterpreterState``
+for that type.
+
+
+Reference Implementation
+========================
+
+<TBD>
+
+
+Open Issues
+===========
+
+* what are the risks/hurdles involved with moving the allocators?
+* is "allow_all_extensions" the best name for the context manager?
+
+
+Deferred Functionality
+======================
+
+* ``PyInterpreterConfig`` option to always run the interpreter in a new thread
+* ``PyInterpreterConfig`` option to assign a "main" thread to the interpreter
+  and only run in that thread
+
+
+Rejected Ideas
+==============
+
+<TBD>
+
+
+References
+==========
+
+Related:
+
+* :pep:`384`
+* :pep:`432`
+* :pep:`489`
+* :pep:`554`
+* :pep:`573`
+* :pep:`587`
+* :pep:`630`
+* :pep:`683`
+* :pep:`3121`
+
+
+Copyright
+=========
+
+This document is placed in the public domain or under the
+CC0-1.0-Universal license, whichever is more permissive.
+
+
+..
+    Local Variables:
+    mode: indented-text
+    indent-tabs-mode: nil
+    sentence-end-double-space: t
+    fill-column: 70
+    coding: utf-8
+    End:
```

Comment:
```suggestion
```

No need for this anymore per the current template/guidance (unless you're an Emacs users who doesn't use `rst-mode`).